src/HOFs/propFormatter.js

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react'
+import { PARAMS_EXP_MAP } from "./constants"
+
+/**
+ * Creates a 1-step, or complete, URL from `domain` and `src` Strings.
+ *
+ * - First, the function checks if src has a defined `domain`. If it does, it 
+ * checks to see if `src` has a scheme, and prepends "http" or "https" as needed
+ * - Otherwise, formatSrc formats `domain` and `src` Strings.
+ *   - First it strips the two strings of the  leading and `/` or trailing `/` 
+ *     slash characters.
+ *   - Then, it joins the two strings on a `/` character. IE, 
+ *    `strippedDomain + "/" + strippedSrc`.
+ *   - If `domain` String argument `null` or `undefined`, the function returns
+ *    the original `src` String.
+ *
+ * @param {String} src - URL that is either 1-step or 2-step
+ * @param {String} domain - Domain string, optional
+ * @returns 1-step, or complete, URL String. Ex, _assets.ix.net/foo/bar.jpg_
+ */
+export function formatSrc(src, domain, useHTTPS = true) {
+  // ignore if already has protocol
+  if (src.indexOf("://") !== -1){
+    return src
+  } else {
+    // prepend domain if defined
+    if (domain == null) {
+      return src
+    }
+    let strippedDomain = domain ? domain.replace(/^\/|\/$/g, '') : ""
+    let strippedSrc = src.replace(/^\/|\/$/g, '')
+    return "https://" + strippedDomain + "/" + strippedSrc;
+  }
+}
+
+/**
+ * A function that formats the following values in the props Object:
+ *
+ * - `width`: if undefined or negative gets set to `null`.
+ * - `height`: if undefined or negative gets set to `null`.
+ * - `src`: concatenated to `domain` if `src` defined and has no domain.
+ *
+ * @param {Object} props 
+ * @returns A formatted `props` Object.
+ */
+export const formatProps = (props) => {
+  const width = !props.width || props.width <= 1 ? null : props.width
+  const height = !props.height || props.height <= 1 ? null : props.height
+  const src = props.src ? formatSrc(props.src, props.domain) : null
```

Comment:
I like this, but I really try avoid using `null` unless necessary, and much prefer `undefined`. What do you think?